### ISSUE TITLE
Error: package or namespace load failed

### ISSUE THREAD

**Issue by VectorPosse** [ProjectMOSAIC/ggformula#108]
I just upgraded to  R 3.5.1 and loading mosaic with `library` breaks with this error:

`Error: package or namespace load failed for ‘ggformula’ in namespaceExport(ns, exports):
 undefined exports: %s!=%, %s!==%, %s+%, %s<%, %s<=%, %s==%, %s===%, %s>%, %s>=%, %stri!=%, %stri!==%, %stri+%, %stri<%, %stri<=%, %stri==%, %stri===%, %stri>%, %stri>=%, stri_datetime_add<-, stri_sub<-, stri_subset<-, stri_subset_charclass<-, stri_subset_coll<-, stri_subset_fixed<-, stri_subset_regex<-, stri_c, stri_c_list, stri_cmp, stri_cmp_eq, stri_cmp_equiv, stri_cmp_ge, stri_cmp_gt, stri_cmp_le, stri_cmp_lt, stri_cmp_neq, stri_cmp_nequiv, stri_compare, stri_conv, stri_count, stri_count_boundaries, stri_count_charclass, stri_count_coll, stri_count_fixed, stri_count_regex, stri_count_words, stri_datetime_add, stri_datetime_create, stri_datetime_fields, stri_datetime_format, stri_datetime_fstr, stri_datetime_now, stri_datetime_parse, stri_datetime_symbols, stri_detect, stri_detect_charclass, stri_detect_coll, stri_detect_fixed, stri_detect_regex, stri_dup, stri_duplicated, stri_duplicated_any, stri_enc_detect, stri_enc_detect2, stri_enc_fromutf32, stri_enc_get, stri_enc_info, stri_enc_isascii, stri_enc_isutf16be, stri_enc_isutf16le, stri_enc_isutf32be, stri_enc_isutf32le, stri_enc_isutf8, stri_enc_list, stri_enc_mark, stri_enc_set, stri_enc_toascii, stri_enc_tonative, stri_enc_toutf32, stri_enc_toutf8, stri_encode, stri_endswith, stri_endswith_charclass, stri_endswith_coll, stri_endswith_fixed, stri_escape_unicode, stri_extract, stri_extract_all, stri_extract_all_boundaries, stri_extract_all_charclass, stri_extract_all_coll, stri_extract_all_fixed, stri_extract_all_regex, stri_extract_all_words, stri_extract_first, stri_extract_first_boundaries, stri_extract_first_charclass, stri_extract_first_coll, stri_extract_first_fixed, stri_extract_first_regex, stri_extract_first_words, stri_extract_last, stri_extract_last_boundaries, stri_extract_last_charclass, stri_extract_last_coll, stri_extract_last_fixed, stri_extract_last_regex, stri_extract_last_words, stri_flatten, stri_info, stri_isempty, stri_join, stri_join_list, stri_length, stri_list2matrix, stri_locale_get, stri_locale_info, stri_locale_list, stri_locale_set, stri_locate, stri_locate_all, stri_locate_all_boundaries, stri_locate_all_charclass, stri_locate_all_coll, stri_locate_all_fixed, stri_locate_all_regex, stri_locate_all_words, stri_locate_first, stri_locate_first_boundaries, stri_locate_first_charclass, stri_locate_first_coll, stri_locate_first_fixed, stri_locate_first_regex, stri_locate_first_words, stri_locate_last, stri_locate_last_boundaries, stri_locate_last_charclass, stri_locate_last_coll, stri_locate_last_fixed, stri_locate_last_regex, stri_locate_last_words, stri_match, stri_match_all, stri_match_all_regex, stri_match_first, stri_match_first_regex, stri_match_last, stri_match_last_regex, stri_na2empty, stri_numbytes, stri_opts_brkiter, stri_opts_collator, stri_opts_fixed, stri_opts_regex, stri_order, stri_pad, stri_pad_both, stri_pad_left, stri_pad_right, stri_paste, stri_paste_list, stri_rand_lipsum, stri_rand_shuffle, stri_rand_strings, stri_read_lines, stri_read_raw, stri_remove_empty, stri_replace, stri_replace_all, stri_replace_all_charclass, stri_replace_all_coll, stri_replace_all_fixed, stri_replace_all_regex, stri_replace_first, stri_replace_first_charclass, stri_replace_first_coll, stri_replace_first_fixed, stri_replace_first_regex, stri_replace_last, stri_replace_last_charclass, stri_replace_last_coll, stri_replace_last_fixed, stri_replace_last_regex, stri_replace_na, stri_reverse, stri_sort, stri_split, stri_split_boundaries, stri_split_charclass, stri_split_coll, stri_split_fixed, stri_split_lines, stri_split_lines1, stri_split_regex, stri_startswith, stri_startswith_charclass, stri_startswith_coll, stri_startswith_fixed, stri_stats_general, stri_stats_latex, stri_sub, stri_subset, stri_subset_charclass, stri_subset_coll, stri_subset_fixed, stri_subset_regex, stri_timezone_get, stri_timezone_info, stri_timezone_list, stri_timezone_set, stri_trans_char, stri_trans_general, stri_trans_isnfc, stri_trans_isnfd, stri_trans_isnfkc, stri_trans_isnfkc_casefold, stri_trans_isnfkd, stri_trans_list, stri_trans_nfc, stri_trans_nfd, stri_trans_nfkc, stri_trans_nfkc_casefold, stri_trans_nfkd, stri_trans_tolower, stri_trans_totitle, stri_trans_toupper, stri_trim, stri_trim_both, stri_trim_left, stri_trim_right, stri_unescape_unicode, stri_unique, stri_width, stri_wrap, stri_write_lines
Error: package ‘ggformula’ could not be loaded`

I've tried uninstalling and reinstalling both mosaic and ggformula.

My students have a test tomorrow, so this is a somewhat urgent problem.

**Comment by nicholasjhorton** [comment]
Sorry to hear about the complication. 

Can you try installing the development version of ggformula from GitHub?  

See GitHub.com/ProjectMOSAIC/ggformula for details. 

All the best,

Nick

**Comment by VectorPosse** [comment]
That's a great suggestion, but, alas, it also fails with the same error.

**Comment by VectorPosse** [comment]
I just uploaded my .Rmd to our instance of RStudio Server and it seems to be working. (It was updated to 3.5.1 last week, so I'm not sure what broke between then and now.) So crisis averted for my students at least. But I still can't run mosaic commands on my own machine.

**Comment by nicholasjhorton** [comment]
Can you please include the output of sessionInfo() after you get this error?

All the best,

Nick

**Comment by VectorPosse** [comment]
No problem:

```
R version 3.5.1 (2018-07-02)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] devtools_1.13.6 ggplot2_3.0.0   lattice_0.20-35 dplyr_0.7.6     car_3.0-0      
[6] carData_3.0-1   broom_0.5.0    

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.18      tidyr_0.8.1       assertthat_0.2.0  rprojroot_1.3-2  
 [5] digest_0.6.15     R6_2.2.2          cellranger_1.1.0  plyr_1.8.4       
 [9] backports_1.1.2   httr_1.3.1        pillar_1.3.0      rlang_0.2.1      
[13] lazyeval_0.2.1    curl_3.2          readxl_1.1.0      rstudioapi_0.7   
[17] data.table_1.11.4 foreign_0.8-70    munsell_0.5.0     compiler_3.5.1   
[21] pkgconfig_2.0.1   htmltools_0.3.6   tidyselect_0.2.4  tibble_1.4.2     
[25] mosaicCore_0.6.0  rio_0.5.10        crayon_1.3.4      withr_2.1.2      
[29] MASS_7.3-50       grid_3.5.1        nlme_3.1-137      gtable_0.2.0     
[33] git2r_0.23.0      magrittr_1.5      scales_0.5.0      zip_1.0.0        
[37] bindrcpp_0.2.2    openxlsx_4.1.0    tools_3.5.1       forcats_0.3.0    
[41] glue_1.3.0        purrr_0.2.5       hms_0.4.2         abind_1.4-5      
[45] yaml_2.2.0        colorspace_1.3-2  memoise_1.1.0     knitr_1.20       
[49] bindr_0.1.1       haven_1.1.2      
```

**Comment by nicholasjhorton** [comment]
Is this relevant?

https://community.rstudio.com/t/error-package-or-namespace-load-failed-for-tidytext-in-namespaceexport-ns-exports/10391

All the best,

Nick

**Comment by VectorPosse** [comment]
Hmm...I also get that error when I try to load tidytext, but I also still
get it with mosaic/ggformula. I've tried removing and reinstalling tidytext
as well, but that didn't fix it. This is starting to look like something
caused by yet another package, but I'm not sure which.

I really appreciate you looking into this. Sorry it's turning out to be
sticky.

**Comment by VectorPosse** [comment]
(To be clear, they are slightly different errors. My tidytext error matches
the one at the link you sent, but it is not as long as the ggformula error.)

**Comment by VectorPosse** [comment]
I just installed the `stringi` package, and now everything works. (I should have Googled the commands listed in the error message I suppose. Still not sure why ggformula and tidytext are getting hung up on that. If `stringi` is a dependency, shouldn't those packages try to install it?)

**Comment by rpruim** [comment]
You might try
```r
update.packages(checkBuilt = TRUE)
```
That will make sure that will make sure that all of the dependencies not only exist but were built with the current version of R.  I've found that to be very helpful when updating to a new version of R.  That would likely have caused a rebuild of `stringi`.

The `stringi` dependency comes via `stringr` which is imported by `ggformula`.  If I had to guess, when `ggformula` was updated, it appeared that `stringr` was there and good to go, but when actually attempting to use `stringr`, the problems with `stringi` were detected.  I think dependency checks only look for existence (or version number if that is required by the package DESCRIPTION) and not whether the packages work under the current version of R.
